imgee/tasks.py

```diff
@@ -1,35 +1,85 @@
+import re
 import redis
-from imgee import app
+
+
+class InvalidRedisQueryException(Exception):
+    pass
 
 
 class TaskRegistry(object):
-    def __init__(self, name='default', connection=None):
-        self.connection = redis.from_url(connection) if connection else None
+    def __init__(self, name='default', connection=None, app=None):
         self.name = name
-        self.key = 'imgee:registry:%s' % name
+        self.connection = connection
+        self.key_prefix = 'imgee:registry:%s' % name
+        self.filename_pattern = '^[a-z0-9\_\.]+$'
```

Comment:
Did you mean to use `re.compile(r'^[a-z0-9_\.]+$')`? It needs to be a raw `r` string to remove ambiguity around the `\`, and the `.` needs escaping in regex, but not `_`.